New Checkout Drop-in v1.4.0-beta2 (#273)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/checkout/containers/shipping-methods.mdx

import OptionsTable from '@components/OptionsTable.astro';

The `ShippingMethods` container is designed to manage and display the selection of available shipping methods during the checkout process. You can configure it to handle the selection of shipping methods, display the available shipping methods, and manage the main slot for the shipping methods.

> This container contains internal functionality to hide itself in case the cart is empty or virtual.

Reviewer: ```suggestion
This container includes internal functions to hide itself if the cart is empty or virtual.
```

export interface ShippingMethodsProps extends HTMLAttributes<HTMLDivElement>, TitleProps {
  onShippingMethodSelect?: (method: ShippingMethod) => void;
  preSelectedMethod?: { carrierCode: string; methodCode: string };
  UIComponentType?: UIComponentType;
}
```

> <li>The `displayTitle (*)` property is inherited from `TitleProps` interface; this is used to decide whether the title is displayed or not.</li>

Reviewer: ```suggestion
<li>The `displayTitle (*)` property is inherited from the `TitleProps` interface. It is used to determine whether to display the title.</li>
```

<li>The `onShippingMethodSelect` property is a handler used to perform actions called when a shipping method is selected.</li>

<li>The `preSelectedMethod` property is an object containing the pair of _carrierCode_ and _methodCode_ values which defines the pre-selected shipping method to show.</li>

> <li>The `UIComponentType` property is a string containing the name of the UI component type to be used as a selector for each shipping method. The UI components available are: 'ToggleButton' or 'RadioButton'.</li>

Reviewer: ```suggestion
<li>The `UIComponentType` property is a string containing the name of the UI component type to be used as a selector for each shipping method. The available UI components are: `ToggleButton` or `RadioButton`.</li>
```

> <li>The `slots (*)` property is inherited from `TitleProps` interface; this is an object containing the following properties:</li>

Reviewer: ```suggestion
<li>The `slots (*)` property is inherited from the `TitleProps` interface. It is an object that containing the following properties:</li>
```

> #### Title property (*)

Reviewer: This should actually be another list item nested under the item on line 52.

The `Title` property is a handler used to render a custom title content.

## Example

> The following example renders the `ShippingMethods` container on a checkout page. It includes configurations to hide the title, pre-select the free shipping in case no shipping method is selected, and showing a message in case the chosen shipping method is `Best Way` (Table Rate).

Reviewer: ```suggestion
The following example renders the `ShippingMethods` container on a checkout page. It includes configurations to hide the title, pre-select the free shipping option if no shipping method is selected, and show a message if the chosen shipping method is `Best Way` (Table Rate).
```
